tests/js/server/aql/aql-graph-traverser.js

```diff
@@ -2412,6 +2412,320 @@ function complexFilteringSuite() {
+      let query = `WITH ${vn} FOR v,e,p IN 1..10 OUTBOUND @start @eCol PRUNE pruneCondition = ${condition} OPTIONS {bfs: true} FILTER pruneCondition RETURN {from: e._from, to: e._to}`;
+      let bindVars = {
+        'eCol': en,
+        'start': vertex.A
+      };
+      let queryExplain = AQL_EXPLAIN(query, bindVars).plan.nodes; //check for extra LET
+      let foundExtraLet = false;
+      queryExplain.forEach((node) => { 
+        if(node.type === "CalculationNode" && node["outVariable"]["name"] === "pruneCondition") {
+          foundExtraLet = true;
+        }
+        if(node.type === "FilterNode") {
+          assertEqual(node["inVariable"]["name"], "pruneCondition");
+        }
+        if(node.type === "TraversalNode") {
+          assertEqual(node["options"]["order"], "bfs");
+        }
+      });
+      assertTrue(foundExtraLet);
+    },
+
+    testStorePruneConditionPathInVariableExplain: function () {
+      // note: using NOOPT(...) here to disable the optimization that pulls
+      // certain FILTER conditions into the traversal
+      const condition = "NOOPT(p.edges[0].right) == true";
+      let query = `WITH ${vn} FOR v,e,p IN 1..10 OUTBOUND @start @eCol PRUNE pruneCondition = ${condition} FILTER pruneCondition RETURN {from: e._from, to: e._to}`;
+      let bindVars = {
+        'eCol': en,
+        'start': vertex.A
+      };
+      let queryExplain = AQL_EXPLAIN(query, bindVars).plan.nodes; //check for extra LET
+      let foundExtraLet = false;
+      queryExplain.forEach((node) => { 
+        if(node.type === "CalculationNode" && node["outVariable"]["name"] === "pruneCondition") {
+          foundExtraLet = true;
+        }
+        if(node.type === "FilterNode") {
+          assertEqual(node["inVariable"]["name"], "pruneCondition");
+        }
+      });
+      assertTrue(foundExtraLet);
+    },
+
+    testStorePruneConditionPathInVariableExplainWithOptions: function () {
+      // note: using NOOPT(...) here to disable the optimization that pulls
+      // certain FILTER conditions into the traversal
+      const condition = "NOOPT(p.edges[0].right) == true";
+      let query = `WITH ${vn} FOR v,e,p IN 1..10 OUTBOUND @start @eCol PRUNE pruneCondition = ${condition} OPTIONS {bfs: true} FILTER pruneCondition RETURN {from: e._from, to: e._to}`;
+      let bindVars = {
+        'eCol': en,
+        'start': vertex.A
+      };
+      let queryExplain = AQL_EXPLAIN(query, bindVars).plan.nodes; //check for extra LET
+      let foundExtraLet = false;
+      queryExplain.forEach((node) => { 
+        if(node.type === "CalculationNode" && node["outVariable"]["name"] === "pruneCondition") {
+          foundExtraLet = true;
+        }
+        if(node.type === "FilterNode") {
+          assertEqual(node["inVariable"]["name"], "pruneCondition");
+        }
+        if(node.type === "TraversalNode") {
+          assertEqual(node["options"]["order"], "bfs");
+        }
+      });
+      assertTrue(foundExtraLet);
+    },
+
```

Comment:
I like the above tests, very good, they over a lot.

However I would like to ask for some more tests, just to increase the confidence here.

1) Reassigning of variables:
  a) Add a variable, then try to reassign it as the "prune" variabel (you did this on `p` however this `p` is part of the node and somewhat special. I am looking more for `FOR other IN 1..2 FOR v IN 1 OUTBOUND ... PRUNE other = v.name == "test"`
    b) Try to resign the prune condition later `PRUNE pruneCondition = v.name == "test" LET pruneCondition = e.weight`
    c) Use a second traversal statement, and try to reassign the prune condition there:
       `FOR v1 IN <...> PRUNE pruneCondition = v1.name == "test"  <...> FOR v2 IN 1 OUTBOUND v1 PRUNE pruneCondition = v2.name`

All of the above tests should fail due to invalid reassignment.
2) Subquery
  Try to add PRUNE pruneConditon = (FOR other IN 1..2 RETURN other)
  This should be disallowed as well. (subquery is disallowed)
3) Use the pruneCondition in more complex places (I expect all of those to work)
  a) Try to use it within a subquery `PRUNE condition = .... LET subQuery = (FOR q IN p.edges FILTER condition RETURN q)`
   b) Try to use it in a calculation LET array = [] FOR v IN 1 OUTBOUND PRUNE condition = <....> RETURN PUSH(array, condition)
  c) Try to use a second Traversal and use the Prune Expression within a path filter:
  ```
   FOR v1 IN 1 OUTBOUND <...> PRUNE pruneCondition = v1.name == "test"
    FOR v2, e2, p2 IN 2 ANY v1 FILTER  p2.edges[*].value ALL== pruneCondition
```

All of the above tests basically target the fact that the calculation is handed around, and we need to make sure the references stay intact.
If I understood the implementation correctly you are not storing the result (Which you cannot get to) but your are storing the way it should be calculated.

Reply:
Added these tests in the last commit